crates/tests/registry/tests/bytes.rs

```diff
@@ -0,0 +1,24 @@
+use windows_registry::*;
+
+#[test]
+fn bytes() -> Result<()> {
+    let test_key = "software\\windows-rs\\tests\\bytes";
+    _ = CURRENT_USER.remove_tree(test_key);
+    let key = CURRENT_USER.create(test_key)?;
+
+    key.set_bytes("bytes", Type::Bytes, &[1, 2, 3])?;
+    assert_eq!(key.get_type("bytes")?, Type::Bytes);
```

Comment:
The latter works - retrieve `Value` and then convert as needed - but there is some benefit to having things like `get_u32` since they can avoid a heap allocation and `get_hstring` can avoid a second allocation.

Reply:
I did try to get there for this reason but alas it's just not as efficient. 

Also, all of the `get`/`set` methods deal with values so it seems unnecessarily verbose to keep saying `get_u32_value` instead of just `get_u32` when it's pretty obvious what you're asking for.

Reply:
If we add last write timestamps, security descriptors, etc. support to the keys, we then end up with a mixed key/value API surface, as such:

```rust
key.get_u32
key.get_security_descriptor
key.get_timestamp
```

Is that ok?

Reply:
I think that works because:

* Those key properties aren't registry value types
* The value queries are named `get_u32("name")` vs `get_timestamp()`

Reply:
That's fair. Definitely not worth breaking changes in this area yet.